### DISCUSSION TITLE
Feature addition

### DISCUSSION THREAD

**Discussion by gouravk1226** [golang-jwt/jwt#285] [Ideas]
can we add these feature?
1. token blacklisting
2. one function that would take login credential as a parameter and return access and refresh token as an output. 
the idea behind this feature is to easy the development for the developer. I am referring developing something similar to django jwt.

**Comment by oxisto** [comment]
Hi!

In general we are trying to limit the scope of this library to a core set of functions. The primary reason for this is that this library is used by A LOT of packages in the Go ecosystem and there only a handful people that are actively maintaining it. So we need to weigh up, if a new feature is really wanted by a lot of people.

1) Token blacklisting. Not quite sure if I understand this 100 % correct, but blocking a specific token is just basically a string operation, so I am not quite sure if this should be handled by a jwt library. Probably it should rather be handled by a JWT middleware such as https://github.com/auth0/go-jwt-middleware.

2) If I understand it correctly, you are referring to an oauth flow with client credentials to retrieve an access/refresh token? This is actually handled by the [golang.org/x/oauth2](https://golang.org/x/oauth2) package, which is an "x" additional to the Go standard library. They have a good example here (https://pkg.go.dev/golang.org/x/oauth2#example-Config) how to "exchange" client credentials (client ID / client secret) with an access/refresh token.
